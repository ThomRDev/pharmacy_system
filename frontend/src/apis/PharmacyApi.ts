import axios from "axios"

export const pharmacyApi = axios.create({
  baseURL : "http://localhost:8000/api",
  // baseURL : "https://farmaciabackend-production.up.railway.app/api",
  headers : {
    "Content-Type" :"application/json",
  },
  withCredentials : true
})

// roles
// https://github.com/gitdagray/react_protected_routes/blob/main/src/components/RequireAuth.js