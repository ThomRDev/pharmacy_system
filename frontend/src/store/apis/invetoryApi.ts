import { createApi,fetchBaseQuery } from "@reduxjs/toolkit/query/react"
export const inventoryApi = createApi({
  reducerPath : "inventory",
  baseQuery : fetchBaseQuery({
    baseUrl : "http://localhost:8000/api",
    // baseUrl : "https://farmaciabackend-production.up.railway.app/api",
    prepareHeaders :(headers) => {
      const token = localStorage.getItem("token")
      headers.set('Content-Type', `application/json`)
      if (token) headers.set('Authorization', `Bearer ${token}`)
      return headers
    }
  }),
  keepUnusedDataFor: 15,
  refetchOnMountOrArgChange: 15,
  endpoints: (builder) => ({
    getProducts : builder.query({
      query: () => "/products"
    }),
    getProductById : builder.query({
      query: (productID:string) => {
        return "/products/product/"+productID
      }
    }),

  })
})

export const { useGetProductsQuery,useGetProductByIdQuery } = inventoryApi