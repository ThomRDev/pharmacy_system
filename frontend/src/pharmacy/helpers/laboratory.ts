import { AxiosError } from "axios"
import { pharmacyApi } from "../../apis"

export const createNewLaboratory = async (laboratory:any) => {
  try{
    const { data } = await pharmacyApi.post("/laboratories/create",{
      ...laboratory
    },{
      headers:{
        "Authorization" : `Bearer ${localStorage.getItem("token")}`
      }
    })
    return data
  }catch(err){
    return (err as AxiosError).response?.data
  }
}

export const getLaboratoryById = async (id:string) => {
  try {
    const { data } = await pharmacyApi.get("/laboratories/laboratory/"+id,{
      headers:{
        "Authorization" : `Bearer ${localStorage.getItem("token")}`
      }
    })
    return data
  } catch (error) {
    return (error as AxiosError ).response?.data
  }
}

export const updateLaboratory = async (laboratory:any) => {
  try{
    const { data } = await pharmacyApi.put("/laboratories/update/"+laboratory.id,{
      ...laboratory,
    },{
      headers:{
        "Authorization" : `Bearer ${localStorage.getItem("token")}`
      }
    })
    return data
  }catch(err){
    return (err as AxiosError ).response?.data
  }
}

export const getLaboratories = async () => {
  try {
    const { data } = await pharmacyApi.get("/laboratories",{
      headers:{
        "Authorization" : `Bearer ${localStorage.getItem("token")}`
      }
    })
    return data
  } catch (error) {
    return (error as AxiosError ).response?.data
  }
}