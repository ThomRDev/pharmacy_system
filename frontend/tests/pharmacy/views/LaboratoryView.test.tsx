

import { expect } from '@jest/globals';
import { configureStore } from '@reduxjs/toolkit';
import { render, screen } from '@testing-library/react';
import "jest";
import { Provider } from 'react-redux';
import { LaboratoryView } from '../../../src/pharmacy/views/LaboratoryView';
import { inventoryApi } from '../../../src/store/apis/invetoryApi';
import { laboratoryApi, useGetLaboratoriesQuery } from '../../../src/store/apis/laboratoryApi';
import { laboratorySlice } from '../../../src/store/laboratory/laboratorySlice';
// jest.mock("../../../src/store/apis/laboratoryApi")
const mockUseGetLaboratoriesQuery = jest.fn()

const store = configureStore({
  reducer :{
    laboratoryStore : laboratorySlice.reducer,
    [inventoryApi.reducerPath] : inventoryApi.reducer,
    [laboratoryApi.reducerPath] : laboratoryApi.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(inventoryApi.middleware).concat(laboratoryApi.middleware)
})

const laboratories = [
  {
    "id": "abc16638-f775-4ceb-a40d-0b889666b9c2",
    "createdAt": "2022-07-24T11:54:56.632Z",
    "updatedAt": "2022-07-24T11:54:56.632Z",
    "address": "Lima Peru",
    "description": "Laboratorio Muerte description",
    "name": "Laboratorio Muerte",
    "lng": "-73.990593",
    "lat": "40.740121"
  },
  {
    "id": "b003453e-0fb7-11ed-91ed-e0d55eb315bd",
    "createdAt": "2022-07-30T13:27:36.000Z",
    "updatedAt": "2022-07-30T13:27:36.000Z",
    "address": "Argentina, Buenos Aires",
    "description": "Laboratorio en Argentina",
    "name": "ArgentinaLab",
    "lng": "-77.02053535953974",
    "lat": "-11.942631145218613"
  }
];

jest.mock("../../../src/store/apis/laboratoryApi.ts",()=>({
  ...jest.requireActual("../../../src/store/apis/laboratoryApi.ts"),
  useGetLaboratoriesQuery : () => mockUseGetLaboratoriesQuery()
}))

describe('Testing al componente de <LaboratoryView />', () => {
  beforeEach(() => jest.clearAllMocks());
  test('Deberia hacer match con el snapshot', () => {
    mockUseGetLaboratoriesQuery.mockReturnValue({
      refetch: jest.fn(),
      isLoading : false,
      data :{
        data : laboratories
      }
    })
    const { container } = render(
      <Provider store={store}>
        <LaboratoryView />
      </Provider> 
    )
    expect(container).toMatchSnapshot();
  })
  test('Deberia renderizar la tabla de los laboratorios', () => {
    mockUseGetLaboratoriesQuery.mockReturnValue({
      refetch: jest.fn(),
      isLoading : false,
      data :{
        data : laboratories
      }
    })
    render(
      <Provider store={store}>
        <LaboratoryView />
      </Provider> 
    )
    expect( screen.getAllByText('Laboratorios').length ).toBeGreaterThanOrEqual(1);
    expect( screen.getByText('Laboratorio Muerte description') ).toBeTruthy()
    expect( screen.getByText('Laboratorio en Argentina') ).toBeTruthy()
    expect( screen.getByLabelText('tabla-laboratorios') ).toBeTruthy()
  })
})