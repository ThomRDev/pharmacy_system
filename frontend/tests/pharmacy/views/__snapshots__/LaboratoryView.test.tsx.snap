// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing al componente de <LaboratoryView /> Deberia hacer match con el snapshot 1`] = `
<div>
  <div
    style="height: 400px; width: 100%;"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 css-1h5yuy8-MuiTypography-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1e3x312-MuiSvgIcon-root"
        data-testid="BiotechIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 19c-1.1 0-2 .9-2 2h14c0-1.1-.9-2-2-2h-4v-2h3c1.1 0 2-.9 2-2h-8c-1.66 0-3-1.34-3-3 0-1.09.59-2.04 1.46-2.56C8.17 9.03 8 8.54 8 8c0-.21.04-.42.09-.62C6.28 8.13 5 9.92 5 12c0 2.76 2.24 5 5 5v2H7z"
        />
        <path
          d="M10.56 5.51C11.91 5.54 13 6.64 13 8c0 .75-.33 1.41-.85 1.87l.59 1.62.94-.34.34.94 1.88-.68-.34-.94.94-.34-2.74-7.53-.94.34-.34-.94-1.88.68.34.94-.94.35.56 1.54z"
        />
        <circle
          cx="10.5"
          cy="8"
          r="1.5"
        />
      </svg>
       Laboratorios
    </h3>
    <div
      class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium  css-oi3y1-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
         Agregar un nuevo laboratorio
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      aria-colcount="6"
      aria-label="tabla-laboratorios"
      aria-multiselectable="false"
      aria-rowcount="3"
      class="MuiDataGrid-root MuiDataGrid-root--densityComfortable css-1p1xowz-MuiDataGrid-root"
      role="grid"
    >
      <div>
        <div>
          <div
            class="MuiDataGrid-toolbarContainer css-1j9kmqg-MuiDataGrid-toolbarContainer"
          >
            <button
              aria-label="Select columns"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ColumnIconIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 5H3c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zm14 0h-3c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zm-7 0h-3c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h3c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1z"
                  />
                </svg>
              </span>
              Columns
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="Show filters"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <span
                  class="MuiBadge-root BaseBadge-root css-1c32n2y-MuiBadge-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FilterListIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                    />
                  </svg>
                  <span
                    class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorPrimary BaseBadge-badge BaseBadge-invisible css-1pi4uwz-MuiBadge-badge"
                  >
                    0
                  </span>
                </span>
              </span>
              Filters
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls=":r2:"
              aria-haspopup="menu"
              aria-label="Density"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              id=":r1:"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ViewStreamIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 18h17v-6H4v6zM4 5v6h17V5H4z"
                  />
                </svg>
              </span>
              Density
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-haspopup="menu"
              aria-label="Export"
              aria-labelledby=":r4:"
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
              id=":r3:"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SaveAltIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 12v7H5v-7H3v7c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zm-6 .67l2.59-2.58L17 11.5l-5 5-5-5 1.41-1.41L11 12.67V3h2z"
                  />
                </svg>
              </span>
              Export
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="MuiBox-root css-1rr4qq7"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
      >
        <div
          class="MuiDataGrid-columnHeaders css-f3jnds-MuiDataGrid-columnHeaders"
          style="min-height: 72px; max-height: 72px; line-height: 72px;"
        >
          <div
            aria-rowindex="1"
            class="MuiDataGrid-columnHeadersInner MuiDataGrid-columnHeadersInner--scrollable css-1s0hp0k-MuiDataGrid-columnHeadersInner"
            role="row"
            style="transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-colindex="1"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-columnHeader--sortable"
              data-field="name"
              role="columnheader"
              style="width: 250px; min-width: 250px; max-width: 250px;"
              tabindex="0"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Nombre
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-controls=":r7:"
                    aria-haspopup="true"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                    id=":r8:"
                    tabindex="-1"
                    title="Menu"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 72px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="2"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter"
              data-field="edit"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Editar
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-controls=":ra:"
                    aria-haspopup="true"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                    id=":rb:"
                    tabindex="-1"
                    title="Menu"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 72px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="3"
              aria-sort="none"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--alignCenter MuiDataGrid-columnHeader--sortable"
              data-field="description"
              role="columnheader"
              style="width: 300px; min-width: 300px; max-width: 300px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainerContent"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Descripcion
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-controls=":rd:"
                    aria-haspopup="true"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                    id=":re:"
                    tabindex="-1"
                    title="Menu"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                style="min-height: 72px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        >
          <div
            class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
            style="width: 0px; height: auto; margin-top: 72px; overflow-x: hidden;"
          >
            <div
              class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
              style="width: auto; height: 134px; min-height: auto;"
            >
              <div
                class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                style="transform: translate3d(0px, 0px, 0px);"
              >
                <div
                  aria-rowindex="2"
                  aria-selected="false"
                  class="MuiDataGrid-row"
                  data-id="abc16638-f775-4ceb-a40d-0b889666b9c2"
                  data-rowindex="0"
                  role="row"
                  style="max-height: 67px; min-height: 67px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                    data-colindex="0"
                    data-field="name"
                    role="cell"
                    style="min-width: 250px; max-width: 250px; min-height: 67px; max-height: 67px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      Laboratorio Muerte
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                    data-colindex="1"
                    data-field="edit"
                    role="cell"
                    style="min-width: 100px; max-width: 100px; min-height: 67px; max-height: 67px;"
                    tabindex="-1"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-juymx1-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                       
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                        data-testid="EditOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m14.06 9.02.92.92L5.92 19H5v-.92l9.06-9.06M17.66 3c-.25 0-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.2-.2-.45-.29-.71-.29zm-3.6 3.19L3 17.25V21h3.75L17.81 9.94l-3.75-3.75z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                    data-colindex="2"
                    data-field="description"
                    role="cell"
                    style="min-width: 300px; max-width: 300px; min-height: 67px; max-height: 67px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      Laboratorio Muerte description
                    </div>
                  </div>
                </div>
                <div
                  aria-rowindex="3"
                  aria-selected="false"
                  class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                  data-id="b003453e-0fb7-11ed-91ed-e0d55eb315bd"
                  data-rowindex="1"
                  role="row"
                  style="max-height: 67px; min-height: 67px;"
                >
                  <div
                    aria-colindex="1"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                    data-colindex="0"
                    data-field="name"
                    role="cell"
                    style="min-width: 250px; max-width: 250px; min-height: 67px; max-height: 67px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      ArgentinaLab
                    </div>
                  </div>
                  <div
                    aria-colindex="2"
                    aria-colspan="1"
                    class="MuiDataGrid-cell--withRenderer MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                    data-colindex="1"
                    data-field="edit"
                    role="cell"
                    style="min-width: 100px; max-width: 100px; min-height: 67px; max-height: 67px;"
                    tabindex="-1"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-juymx1-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                       
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1d159sf-MuiSvgIcon-root"
                        data-testid="EditOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m14.06 9.02.92.92L5.92 19H5v-.92l9.06-9.06M17.66 3c-.25 0-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.2-.2-.45-.29-.71-.29zm-3.6 3.19L3 17.25V21h3.75L17.81 9.94l-3.75-3.75z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    aria-colindex="3"
                    aria-colspan="1"
                    class="MuiDataGrid-cell MuiDataGrid-cell--textCenter"
                    data-colindex="2"
                    data-field="description"
                    role="cell"
                    style="min-width: 300px; max-width: 300px; min-height: 67px; max-height: 67px;"
                    tabindex="-1"
                  >
                    <div
                      class="MuiDataGrid-cellContent"
                    >
                      Laboratorio en Argentina
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Mui-resizeTriggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
      <div>
        <div
          class="MuiDataGrid-footerContainer css-17jjc08-MuiDataGrid-footerContainer"
        >
          <div />
          <div
            class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                1–2 of 2
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
